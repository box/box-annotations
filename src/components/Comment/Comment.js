// @flow
import * as React from 'react';
import noop from 'lodash/noop';

import CommentComponent from '../../../third-party/components/Comment';

import { PLACEHOLDER_USER } from '../../constants';

type Props = {
    onDelete: Function,
    error?: ActionItemError
} & CommentProps;

class Comment extends React.PureComponent<Props> {
    static defaultProps = {
        isPending: false,
        permissions: {
            can_delete: false,
            can_edit: false
        },
        onDelete: noop
    };

    render() {
        const { id, isPending, error, createdAt, modifiedAt, message, onDelete, permissions, createdBy } = this.props;
        return (
            <CommentComponent
                id={id}
                currentUser={PLACEHOLDER_USER}
                onDelete={onDelete}
                created_at={createdAt}
                created_by={createdBy}
                modified_at={modifiedAt}
                tagged_message={message}
                permissions={{ ...permissions }}
                isEditing={false}
                isPending={isPending}
                error={error}
            />
        );
    }
}

export default Comment;
