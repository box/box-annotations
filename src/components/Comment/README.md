### Description
Component to represent an individual annotation.

### Examples
```jsx
const AnnotationExample = require('examples').AnnotationExample;

<AnnotationExample />
```
