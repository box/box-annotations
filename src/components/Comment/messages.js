/* eslint-disable max-len */
/**
 * @flow
 * @file i18n messages
 * @author Box
 */

import { defineMessages } from 'react-intl';
import type { MessageDescriptor } from 'react-intl';

const messages: { [string]: MessageDescriptor } = defineMessages({
    anonymousUserName: {
        id: 'ba.anonymousUserName',
        description: "Placeholder when the current annotation's user information is unknown",
        defaultMessage: 'Some User',
    },
});

export default messages;
