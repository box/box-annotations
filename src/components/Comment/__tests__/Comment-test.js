import * as React from 'react';
import { shallow } from 'enzyme';

import Comment from '../Comment';

const TIME_STRING_SEPT_27_2017 = '2017-09-27T10:40:41-07:00';

const USER = {
    type: 'user',
    id: '789ghi',
    name: 'Mia Thermopolis',
    email: 'princess@genovia.gov',
};

const comment = {
    id: '123',
    createdAt: TIME_STRING_SEPT_27_2017,
    createdBy: USER,
    message: 'test',
    permissions: {},
};

describe('components/Comment', () => {
    beforeEach(() => {});

    const render = (props = {}) => shallow(<Comment {...comment} {...props} />).dive();

    test('should correctly render comment', () => {
        const wrapper = render();
        expect(wrapper).toMatchSnapshot();
    });
});
