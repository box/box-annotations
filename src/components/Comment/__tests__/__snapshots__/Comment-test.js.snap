// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Comment should correctly render comment 1`] = `
<div
  className="bcs-comment-container"
>
  <div
    className="bcs-comment"
    onBlur={[Function]}
    onFocus={[Function]}
  >
    <Avatar
      className="bcs-comment-avatar"
      user={
        Object {
          "email": "princess@genovia.gov",
          "id": "789ghi",
          "name": "Mia Thermopolis",
          "type": "user",
        }
      }
    />
    <div
      className="bcs-comment-content"
    >
      <div
        className="bcs-comment-headline"
      >
        <UserLink
          className="bcs-comment-user-name"
          data-resin-target="activityfeed-profilelink"
          id="789ghi"
          name="Mia Thermopolis"
        />
        <Tooltip
          constrainToScrollParent={false}
          constrainToWindow={true}
          position="top-center"
          text={
            <FormattedMessage
              defaultMessage="{time, date, full} at {time, time, short}"
              id="be.commentPostedFullDateTime"
              values={
                Object {
                  "time": 1506534041000,
                }
              }
            />
          }
          theme="default"
        >
          <small
            className="bcs-comment-created-at"
          >
            <ReadableTime
              relativeThreshold={3600000}
              timestamp={1506534041000}
            />
          </small>
        </Tooltip>
      </div>
      <CommentText
        id="123"
        tagged_message="test"
        translationEnabled={false}
        translationFailed={null}
      />
    </div>
  </div>
</div>
`;
